File: src/main/java/org/example/kalkulationsprogramm/dto/Artikel/KategorieResponseDto.java
package org.example.kalkulationsprogramm.dto.Artikel;

import lombok.Getter;
import lombok.Setter;
import org.example.kalkulationsprogramm.domain.LieferantRolle;

import java.util.Set;

@Getter
@Setter
public class KategorieResponseDto {
    private Integer id;
    private String bezeichnung;
    private boolean leaf;

    /** Eigene, an dieser Kategorie hinterlegte Rollen (leer = erbt von Oberkategorie). */
    private Set<LieferantRolle> typischeRollen;
}

File: src/main/kotlin/org/example/kalkulationsprogramm/dto/Artikel/KategorieResponseDto.kt
package org.example.kalkulationsprogramm.dto.Artikel

import org.example.kalkulationsprogramm.domain.LieferantRolle

data class KategorieResponseDto(
    var id: Int? = null,
    var bezeichnung: String? = null,
    var leaf: Boolean = false,
    var typischeRollen: Set<LieferantRolle>? = null,
)
